src/common/bpf/mod.rs

```diff
@@ -0,0 +1,101 @@
+use super::*;
+use ouroboros::*;
+use std::os::fd::FromRawFd;
+
+mod counters;
+mod distribution;
+
+use counters::Counters;
+use distribution::Distribution;
+
+const PAGE_SIZE: usize = 4096;
+const CACHELINE_SIZE: usize = 64;
+
+/// The maximum number of CPUs supported. Used to make `CounterSet`s behave like
+/// per-CPU counters by packing counters into cacheline sized chunks such that
+/// no CPUs will share cacheline sized segments of the counter map.
+static MAX_CPUS: usize = 1024;
+
+/// The number of histogram buckets based on a rustcommon histogram with the
+/// parameters `m = 0`, `r = 8`, `n = 64`.
+///
+/// NOTE: this *must* remain in-sync across both C and Rust components of BPF
+/// code.
+const HISTOGRAM_BUCKETS: usize = 7424;
+const HISTOGRAM_PAGES: usize = 15;
+
+/// This function converts indices back to values for rustcommon histogram with
```

Comment:
How soon do you think we may want to introduce smaller histograms, say an order of magnitude or more smaller, either by virtue of much more limited value range, or because we can afford much coarser buckets?

Reply:
I don't see this as an urgent thing. I think consistency here gives us some nice guarantees around precision/error where we can document that it's less than some % and the users can be confident in that. It's only 60KB for the histogram in the BPF program.

The userspace heatmaps might be another story, but I think we can use lower precision or constrain the range there if appropriate.